src/Magento/FunctionalTestingFramework/Allure/Event/AddUniqueAttachmentEvent.php

```diff
@@ -1,35 +1,63 @@
 <?php
-
 /**
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
+
 namespace Magento\FunctionalTestingFramework\Allure\Event;
 
 use Symfony\Component\Mime\MimeTypes;
 use Yandex\Allure\Adapter\AllureException;
 use Yandex\Allure\Adapter\Event\AddAttachmentEvent;
 
-const DEFAULT_FILE_EXTENSION = 'txt';
-const DEFAULT_MIME_TYPE = 'text/plain';
-
 class AddUniqueAttachmentEvent extends AddAttachmentEvent
 {
+    private const DEFAULT_FILE_EXTENSION = 'txt';
+    private const DEFAULT_MIME_TYPE = 'text/plain';
+
     /**
-     * @var string
+     * @var AddUniqueAttachmentEvent|null
      */
-    private $type;
+    private static $instance;
 
     /**
-     * Near copy of parent function, added uniqid call for filename to prevent buggy allure behavior
-     * @param string $filePathOrContents
+     * An alternative way to instantiate an instance of this class. Used to mock this class object in unit tests.
+     *
+     * @param mixed       $filePathOrContents
+     * @param string      $caption
+     * @param string|null $type
+     *
+     * @return AddUniqueAttachmentEvent
+     */
+    public static function getInstance(
+        $filePathOrContents,
+        string $caption,
+        ?string $type = null
+    ): AddUniqueAttachmentEvent {
+        if (!self::$instance) {
+            self::$instance = new AddUniqueAttachmentEvent(
+                $filePathOrContents,
+                $caption,
+                $type
+            );
+        }
+        return self::$instance;
+    }
```

Comment:
This alternative way to instantiate an object of this class is only used for unit tests. This should not be used in MFTF. Am I understand it correctly here? If this is the case, can you make it clear in docblock?

Reply:
It's still confusing to add this static method in this class. Is there any other way?
The particular test is valuable, if this cannot be mocked by PHPUnit, can we turn this into a functional test https://github.com/magento/magento2-functional-testing-framework/tree/develop/dev/tests/functional?

Reply:
If `copyFile()` is the reason that prevents you from mocking the class, you could do something like:
```
private function copyFile($filePath, $outputPath)
{
    if (MftfApplicationConfig::getConfig()->getPhase() == MftfApplicationConfig::UNIT_TEST_PHASE) {
        return true;
    }
    return copy($filePath, $outputPath);
}
```

Reply:
Thank you, @jilu1 ! 
It is a great approach for the copyFile method! I will use it for the testAddAttachmentUniqueName testing method!
For other tested methods we have to mock AddUniqueAttachmentEvent with the singleton approach.

Reply:
Hello @jilu1!
I've done that.
Thank you for your suggestion!